packages/hash/api/src/hashGraph/index.ts

```diff
@@ -0,0 +1,29 @@
+import { DataSource } from "apollo-datasource";
+import { Configuration, GraphApi } from "@hashintel/hash-graph-client";
+import HttpAgent, { HttpsAgent } from "agentkeepalive";
+import axios from "axios";
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+
+const agentConfig = {
+  maxSockets: 128,
+  maxFreeSockets: 20,
+  timeout: 60 * 1000, // ms
+  freeSocketTimeout: 30 * 1000, // ms
+};
+
+const httpAgent = new HttpAgent(agentConfig);
+const httpsAgent = new HttpsAgent(agentConfig);
```

Comment:
Somewhat similar question as before, do we need an HTTPS agent at all? _If_ we swap to HTTPS at some point then I could see us changing the Agent, I don't think we need both

Reply:
We don't need it, no, and if we were to switch to HTTPS without adding the agent here, we'd get pooled HTTP requests but unpooled HTTPS requests. I don't think it hurts to have this here

Reply:
Does an agent add any overhead?

Reply:
I don't suspect it would add any overhead. Perhaps not knowing this is enough of a reason to remove it, but as far as I understand the agents act on a per-host basis, so the pool will only be instantiated on per host as requests are made.